### PR TITLE
Budget App 

### PR DESCRIPTION
**Pull request by Zuhaib042**
### Budget App Complete
**Summary**
- In this milestone, I have done the following:
- Splash screen to have login and signup options.
- Home page (Categories page) where user can add categories, edit and delete them.
  -  for each category user can see its name and the amount of its transactions.
  - by clicking on the category name, the user can be taken to the transactions page where he can add transactions.
- Transactions page where users can do all the crud operations.
  - User can see the total amount of all transactions and can add new transactions by clicking on add button.
- New Category page where users can add a new category by filling out the form.
- New Transaction page where user can add a new category by filling out the form.

### CODE REVIEW

File: db/migrate/20230213110327_create_entities.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class CreateEntities < ActiveRecord::Migration[7.0]
+  def change
+    create_table :entities do |t|
+      t.string :name
+      t.integer :amount
+      t.references :user, null: false, foreign_key: true
+
```
Comment (t-yanick) [inline db/migrate/20230213110327_create_entities.rb:7]: - [x] Great job working on this capstone. It looks awesome. There is an issue we need to look at. From the Entity-Relationship diagram, we have a field of AuthorID which is absent in your entity migration file. If you can recall from the BlogApp project you did in prelude to this capstone, there was a similar E-R diagram in which Author table was not on the E-R diagram but the field of AuthorID was an alias from the User table. Kindly take re-look at this issue

File: db/schema.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+# It's strongly recommended that you check this file into your version control system.
+
+ActiveRecord::Schema[7.0].define(version: 2023_02_14_104102) do
+  # These are extensions that must be enabled in order to support this database
+  enable_extension "plpgsql"
+
+  create_table "entities", force: :cascade do |t|
+    t.string "name"
+    t.integer "amount"
+    t.bigint "user_id", null: false
```
Comment (t-yanick) [inline db/schema.rb]: - [x] I think you should be having AuthorID as per the E-R diagram

File: config/routes.rb [hunk 3]
```diff
@@ -1,6 +1,22 @@
 Rails.application.routes.draw do
+  devise_for :users
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
 
   # Defines the root path route ("/")
-  # root "articles#index"
+  # root "splash_screen#index"
```
Comment (t-yanick) [inline config/routes.rb]: - [x] Kindly remove pieces of commented codes from your codebase before deploying or pushing to Github. This makes your codebase look professional

File: test/models/entity_test.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
+
```
Comment (t-yanick) [inline test/models/entity_test.rb:2]: - [x] Add tests for the various models.

**Review by t-yanick — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED

Hi @Zuhaib042 , 

### Good job so far!

### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Detailed video presentation. I Like your piece of code you are proud of. I am also proud of you
:heavy_check_mark: App has all pages and they work as expected.
:heavy_check_mark: App deployed and works perfectly
:heavy_check_mark: Postgres used and authentication done with Devise
:heavy_check_mark: Followed well the given design guidelines
:heavy_check_mark: Added tests are all passing with errors
:heavy_check_mark: Documentation is very professional

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [x] Your application does not validate the addition of a new category or transaction when the form is not field which is good and respects project requirements, but I think a message to tell the user that those fields cannot be empty will give a better user experience.
- [x] You should add tests for models because I see they are missing

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: test/models/entity_test.rb (hunk 4 above)
Comment (Zuhaib042) [reply test/models/entity_test.rb:2]: - I have written all tests in `spec` folder, not in `test` folder, and I also have mentioned testing command in my readme that in order to run the tests, you can run in terminal `rspec spec`.

**Review by rustam-tolipov — APPROVED** [review]
Hi @Zuhaib042, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Developer-512px-10

## Highlights
1. No Linters errors 💯
2. All requirements are complete 💯

## Optional suggestions
No optional suggestions :)

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**